api/router.js

```diff
@@ -6,31 +6,45 @@ var autoredraw = require("../api/autoredraw")
 
 module.exports = function($window, renderer, pubsub) {
 	var router = coreRouter($window)
+	var identity = function(o){return o}
+	var globalId
+	var routeRequested = false
+	function onbeforeresolve(){routeRequested = true}
+
 	var route = function(root, defaultRoute, routes) {
-		var current = {path: null, component: "div"}
+		var current = {path: null, component: "div", render:identity}
```

Comment:
caching the `render` method means that if one were to modify it on the RouteResolver, it will take a `route.set` call for the new version to become active. Mere `redraw()`s won't be affected.